caas/broker.go

```diff
@@ -278,8 +295,22 @@ type ClusterVersionGetter interface {
 	Version() (*version.Number, error)
 }
 
-// ServiceGetterSetter provides the API to get/set service.
-type ServiceGetterSetter interface {
+// ClusterMetadataGetter provides an API to query cluster metadata.
+type ClusterMetadataGetter interface {
+	// GetClusterMetadata returns metadata about host cloud and storage for the cluster.
+	GetClusterMetadata(storageClass string) (result *ClusterMetadata, err error)
```

Comment:
`ClusterMetadata` is k8s spec thing as well.
Most of the code in `caas/metadata.go` could be moved to `caas/kubernetes`

Reply:
Yes, I wanted to do that initially but there are lots of parts in the code that expect this to be there. I will make a second attempt today.